From c6db626749c056f64d1e3bc9cebdf4db0a6e6a01 Mon Sep 17 00:00:00 2001
From: nda-cunh <hydraldev@gmail.com>
Date: Wed, 23 Jul 2025 23:59:57 +0200
Subject: [PATCH] add other nerdfont icons

---
 plugin/webdevicons.vim | 53 +++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 52 insertions(+), 1 deletion(-)

diff --git a/plugin/webdevicons.vim b/plugin/webdevicons.vim
index 596ec68..24d8f9c 100644
--- a/plugin/webdevicons.vim
+++ b/plugin/webdevicons.vim
@@ -217,7 +217,8 @@ function! s:setDictionaries()
         \ 'cc'       : '',
         \ 'cp'       : '',
         \ 'c'        : '',
-        \ 'cs'       : '',
+        \ 'blp'      : '',
+        \ 'cs'       : '󰌛',
         \ 'h'        : '',
         \ 'hh'       : '',
         \ 'hpp'      : '',
@@ -227,6 +228,46 @@ function! s:setDictionaries()
         \ 'nix'      : '',
         \ 'lua'      : '',
         \ 'java'     : '',
+        \ 'vala'     : '',
+        \ 'water'    : '󰥨',
+        \ 'vapi'     : '',
+        \ 'kt'       : '',
+        \ 'kts'      : '',
+        \ 'ktm'      : '',
+        \ 'ktsm'     : '',
+        \ 'odt'      : '󰏆',
+        \ 'doc'      : '󰏆',
+        \ 'docx'     : '󰏆',
+        \ 'docm'     : '󰏆',
+        \ 'xls'      : '󰏇',
+        \ 'xlsx'     : '󰏇',
+        \ 'pdf'      : '',
+        \ 'zip'      : '',
+        \ 'tar'      : '',
+        \ 'gz'       : '',
+        \ 'bz2'      : '',
+        \ 'asm'      : '',
+        \ 'cfg'      : '',
+        \ 'deb'      : '',
+        \ 'log'      : '',
+        \ 's'        : '',
+        \ 'bin'      : '',
+        \ 'iso'      : '',
+        \ 'ld'       : '',
+        \ 'mp4'      : '',
+        \ 'mkv'      : '',
+        \ 'avi'      : '',
+        \ 'mov'      : '',
+        \ 'mp3'      : '',
+        \ 'wav'      : '',
+        \ 'flac'     : '',
+        \ 'ogg'      : '',
+        \ 'opus'     : '',
+        \ 'm3u'      : '',
+        \ 'suprapack' : '',
+        \ 'xml'      : '󰗀',
+        \ 'svg'      : '',
+        \ 'o'        : '',
         \ 'sh'       : '',
         \ 'fish'     : '',
         \ 'bash'     : '',
@@ -292,6 +333,16 @@ function! s:setDictionaries()
   let s:file_node_exact_matches = {
         \ 'exact-match-case-sensitive-1.txt' : '1',
         \ 'exact-match-case-sensitive-2'     : '2',
+        \ 'a'        : '󰬈',
+        \ 'b'        : '󰬉',
+        \ 'c'        : '󰬊',
+        \ 'd'        : '󰬋',
+        \ 'e'        : '󰬌',
+        \ 'f'        : '󰬍',
+        \ 'x'        : '󰬟',
+        \ 'y'        : '󰬠',
+        \ 'z'        : '󰬡',
+        \ 'meson.build': '',
         \ 'gruntfile.coffee'                 : '',
         \ 'gruntfile.js'                     : '',
         \ 'gruntfile.ls'                     : '',
-- 
2.34.1

